public/data/python.json

```diff
@@ -187,20 +393,153 @@
+          "        full_path = os.path.join(root, file)",
+          "        found_files.append(full_path)",
           "",
-          "    for entry in os.scandir(directory):",
-          "        if entry.is_file() and entry.name.lower().endswith(file_type):",
-          "            found_files.append(entry.name)",
-          "    return found_files",
+          "  return found_files",
           "",
           "# Example Usage:",
           "pdf_files = find_files('/path/to/your/directory', '.pdf')",
           "print(pdf_files)"
         ],
         "tags": ["python", "os", "filesystem", "file_search"],
         "author": "Jackeastern"
+      },
+      {
+        "title": "Append to File",
+        "description": "Appends content to the end of a file.",
+        "code": [
+          "def append_to_file(filepath, content):",
+          "    with open(filepath, 'a') as file:",
+          "        file.write(content + '\\n')",
+          "",
+          "# Usage:",
+          "append_to_file('example.txt', 'This is an appended line.')"
+        ],
+        "tags": ["python", "file", "append", "utility"],
+        "author": "axorax"
+      },
+      {
+        "title": "Check if File Exists",
+        "description": "Checks if a file exists at the specified path.",
+        "code": [
+          "import os",
+          "",
+          "def file_exists(filepath):",
+          "    return os.path.isfile(filepath)",
+          "",
+          "# Usage:",
+          "print(file_exists('example.txt'))  # Output: True or False"
+        ],
+        "tags": ["python", "file", "exists", "check", "utility"],
+        "author": "axorax"
+      },
+      {
+        "title": "Delete File",
+        "description": "Deletes a file at the specified path.",
+        "code": [
+          "import os",
+          "",
+          "def delete_file(filepath):",
+          "    if os.path.exists(filepath):",
+          "        os.remove(filepath)",
+          "        print(f'File {filepath} deleted.')",
+          "    else:",
+          "        print(f'File {filepath} does not exist.')",
+          "",
+          "# Usage:",
+          "delete_file('example.txt')"
+        ],
+        "tags": ["python", "file", "delete", "utility"],
+        "author": "axorax"
+      },
+      {
+        "title": "Copy File",
+        "description": "Copies a file from source to destination.",
+        "code": [
+          "import shutil",
+          "",
+          "def copy_file(src, dest):",
+          "    shutil.copy(src, dest)",
+          "",
+          "# Usage:",
+          "copy_file('example.txt', 'copy_of_example.txt')"
+        ],
+        "tags": ["python", "file", "copy", "utility"],
+        "author": "axorax"
+      },
+      {
+        "title": "Rename File",
+        "description": "Renames a file from its current name to a new name.",
+        "code": [
+          "import os",
+          "",
+          "def rename_file(old_name, new_name):",
```

Comment:
this example has a big flaw, if the file needs a specific path to be accessed it can't be renamed. One option would be to make `old_name` a `path` instead of just the name of the file